modules/upgrade.sh

```diff
@@ -15,7 +15,11 @@ function upgrade {
       fi
       npm install -g '@treehouses/cli@latest'
   else
-      npm install -g "@treehouses/cli@${tag}"
+      if [[ ${tag} =~ [0-9].*\.[0-9].*\.[0-9].*(-\\w).* ]]; then
+        npm install -g "@treehouses/cli@${tag}"
+      else
+        npm install -g "@treehouses/cli@latest";
```

Comment:
I don't get why we need this.
We already can install the latest on L16

can you explain me?

Reply:
There are two possible cases

1. A version tag is not supplied - In this case, we check whether the latest version is installed, and if not, then it installs the latest version as in L16

2. A version tag is supplied - In this case, if the supplied tag follows semver spec, then it is treated as a valid tag, and the cli with specified tag is installed
If the tag turns out to be invalid (doesn't follow semver spec), then it **falls back to install the latest version**